homeassistant/components/coinbase/config_flow.py

```diff
@@ -0,0 +1,211 @@
+"""Config flow for Coinbase integration."""
+import logging
+
+from coinbase.wallet.client import Client
+from coinbase.wallet.error import AuthenticationError
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_API_KEY, CONF_API_TOKEN
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+
+from . import get_accounts
+from .const import (
+    API_ACCOUNT_CURRENCY,
+    API_RATES,
+    CONF_CURRENCIES,
+    CONF_EXCHANGE_RATES,
+    CONF_OPTIONS,
+    CONF_YAML_API_TOKEN,
+    DOMAIN,
+    RATES,
+    WALLETS,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_API_KEY): str,
+        vol.Required(CONF_API_TOKEN): str,
+    }
+)
+
+
+async def validate_api(hass: core.HomeAssistant, data):
+    """Validate the credentials."""
+
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        if entry.data[CONF_API_KEY] == data[CONF_API_KEY]:
+            raise AlreadyConfigured
```

Comment:
We can use this helper:
https://github.com/home-assistant/core/blob/91b4f7d1d53f9614b9a9287696a3346fd2d91d15/homeassistant/config_entries.py#L1128-L1132